euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceByKey.java

```diff
@@ -391,7 +470,8 @@ public void add(E element) {
       if (v == null) {
         this.storage.set(element);
       } else {
-        this.storage.set(this.reducer.apply(Arrays.asList(v, element)));
+        this.reducer.apply(Arrays.asList(v, element), context);
+        this.storage.set(context.getValue());
```

Comment:
Okay, I can change the contract to that instead of `getValue()` a will introduce `getAndResetValue()` which will do as it name suggests. :)